### PR TITLE
doc: Added 54H FLPR documentation

### PR DESCRIPTION
**Pull request by FrancescoSer**
Added 54H FLPR documentation.

### CODE REVIEW

**Comment by FrancescoSer** [comment]
@gmarull @hubertmis Could you check if this is on the right track? Is there anything else that should be mentioned here for lilium? Also, the supported peripheral list should be clarified.

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+When building for the ``nrf54h20dk/nrf54h20/cpuflpr`` target, a minimal sample is automatically loaded onto the application core.
+See more information on :ref:`building_nrf54h_app_flpr_core`.
+
+Memory allocation
+*****************
+
+If a FLPR CPU is running, it can lead to increased latency when accessing ``RAM_21``.
+Because of this, when FLPR is used in a project, you should utilize ``RAM_21`` to store only the FLPR code, FLPR data, and the application CPU's non-time-sensitive information.
+Conversely, you should use ``RAM_0x`` to store data with strict access time requirements such as DMA buffers, and the application CPU data used in low-latency ISRs.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst:41]: @gmarull is RAM0x the correct one here?

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+      .. code-block:: console
+
+         west build -p -b nrf54h20dk/nrf54h20/cpuflpr
+         west flash
+
+   .. group-tab:: Using application that supports multi-image builds
+
+      If your application involves creating custom images for both the application core and the FLPR core, make sure to disable the VPR bootstrapping sample.
+      You can do this by disabling the ``SB_CONFIG_VPR_LAUNCHER`` option when building for the FLPR target.
```
Comment (FrancescoSer) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst:109]: @gmarull Please check if kconfig match, these were originally mentioned for 54L

Comment (gmarull) [reply doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst:109]: they should be the same

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst (hunk 1 above)
Comment (gmarull) [reply doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst:41]: @hubertmis ^

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+   :local:
+   :depth: 2
+
+.. note::
+   The FLPR core support in the |NCS| is currently :ref:`experimental<software_maturity>`.
+
+The nRF54H20 SoC includes a dedicated VPR CPU, based on RISC-V architecture, known as the *fast lightweight peripheral processor* (FLPR).
+The FLPR core can be used to manage specific peripherals through the appropriate Zephyr Device Driver API:
+
+* ##List to be completed
```
Comment (gmarull) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst:16]: uart is for sure supported (UARTE120)

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+When building for the ``nrf54h20dk/nrf54h20/cpuflpr`` target, a minimal sample is automatically loaded onto the application core.
+For more details, see :ref:`building_nrf54h_app_flpr_core`.
+
+Memory allocation
+*****************
+
+Running the FLPR CPU can lead to increased latency when accessing ``RAM_21``.
+To mitigate this, you should use ``RAM_21`` exclusively for FLPR code, FLPR data, and non-time-sensitive information from the application CPU.
+For data that requires strict access times, such as DMA buffers and application CPU data used in low-latency ISRs, you should use ``RAM_0x``.
```
Comment (gmarull) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst:40]: @hubertmis needs your input

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+Separate images
+---------------
+
+You can build and program the application sample and the FLPR sample as separate images using either the |nRFVSC| or the command line.
+To use nRF Util, see `nRF Util`_.
+Depending on the method you select, complete the following steps:
+
+.. tabs::
+
+   .. group-tab:: nRF Connect for VS Code
```
Comment (gmarull) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst:124]: Not familiar with VSCode flow, needs to be validated by someone else

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+   The FLPR core support in the |NCS| is currently :ref:`experimental<software_maturity>`.
+
+The nRF54H20 SoC includes a dedicated VPR CPU, based on RISC-V architecture, known as the *fast lightweight peripheral processor* (FLPR).
+The FLPR core can be used to manage specific peripherals through the appropriate Zephyr Device Driver API:
+
+* ##List to be completed
+
+.. _vpr_flpr_nrf54h20_initiating:
+
+Using FLPR with Zephyr multithreaded mode
```
Comment (hubertmis) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst:20]: I would rephrase the title. Right now it's ambiguous if FLPR or FLPR's owner runs multithreaded Zephyr. How about "Zephyr multithreaded mode on FLPR"? Or "Using Zephyr multithreaded mode on FLPR"?

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst (hunk 4 above)
Comment (hubertmis) [reply doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst:40]:
```suggestion
For data that requires strict access times, such as CPU data used in low-latency ISRs, you should use local RAM or, when greater latency is acceptable, ``RAM_0x``. The DMA buffers should be placed in memory designed to a given peripheral.
```

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+Building for both the application and FLPR core
+===============================================
+
+Building for both the application core and the FLPR core differs from the default |NCS| procedure.
+Additional configuration is required to enable the FLPR core.
+
+This section explains how to build and program both cores, covering separate builds and sysbuild configurations.
+The FLPR core supports two variants:
+
+* ``nrf54h20dk/nrf54h20/cpuflpr``: FLPR runs from SRAM (recommended method).
```
Comment (hubertmis) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst:73]: Being precise here would help to understand the partitioning recommendations described earlier.

```suggestion
* ``nrf54h20dk/nrf54h20/cpuflpr``: FLPR runs from RAM_21 (recommended method).
```

**Comment by divipillai** [comment]
Could add a comment on PR https://github.com/nrfconnect/sdk-nrf/pull/18192, if changelog entry is required.

**Comment by kylebonnici** [comment]
If VSCode is used to build an App using flip as a target build command would be:

Target: nrf54h20dk/nrf54h20/cpuflpr
west build --build-dir <build_dir> <app-dir> --pristine --board nrf54h20dk/nrf54h20/cpuflpr -- -DNCS_TOOLCHAIN_VERSION=NONE

Target: nrf54h20dk/nrf54h20/cpuflpr/xip
west build --build-dir <build_dir> <app-dir> --pristine --board nrf54h20dk/nrf54h20/cpuflpr/xip -- -DNCS_TOOLCHAIN_VERSION=NONE


in NCS 2.8.0-rc2 the only target I see for the nrf54h20dk is nrf54h20dk/nrf54h20/cpuflpr/xip

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst (hunk 3 above)
Comment (FrancescoSer) [reply doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst:16]: @hubertmis @shanthanordic Any additional inputs?

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+
+.. _building_nrf54h:
+
+Building and programming with the nRF54H20 DK
+*********************************************
+
+.. note::
+   The FLPR core support in the |NCS| is currently :ref:`experimental<software_maturity>`.
+
+Depending on the sample, you may need to program only the application core or both the FLPR and application cores.
```
[the remark this answers is not in the document]
Comment (FrancescoSer) [reply doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst:51]: Please recheck

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+
+* ##List to be completed
+
+.. _vpr_flpr_nrf54h20_initiating:
+
+Using FLPR with Zephyr multithreaded mode
+*****************************************
+
+The FLPR core can function as a general-purpose core, running under the full Zephyr kernel.
+Building the FLPR target is similar to building the application core, but the application core build must include an overlay that enables the FLPR core.
```
[the remark this answers is not in the document]
Comment (FrancescoSer) [reply doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst:24]: Please recheck

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+
+##List to be completed
+
+.. _vpr_flpr_nrf54h20_initiating:
+
+Using Zephyr multithreaded mode on FLPR
+***************************************
+
+The FLPR core can operate as a general-purpose core, running under the full Zephyr kernel.
+To enable the FLPR core, always include an overlay in the application core build that configures and builds the FLPR target.
```
Comment (masz-nordic) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst:26]: That is incorrect. The overlay only enables a driver on application core side that starts the FLPR (and copies code for non-xip variant).

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+
+.. _building_nrf54h:
+
+Building and programming with the nRF54H20 DK
+*********************************************
+
+.. note::
+   The FLPR core support in the |NCS| is currently :ref:`experimental<software_maturity>`.
+
+For samples targeting the FLPR core, program the application core, which subsequently programs the FLPR core.
```
Comment (masz-nordic) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst:53]: Not true?

**Review by masz-nordic — DISMISSED** [review]
This description looks like a copy of L15's but out of sync due to reviews in this PR.
Any reason for not combining these two into common part and only listing differences between H20 and L15?

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst (hunk 10 above)
Comment (FrancescoSer) [reply doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst:26]: Shantha asked for this edit @shanthanordic @masz-nordic  Could you please clarify this?

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst (hunk 11 above)
Comment (FrancescoSer) [reply doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst:53]: Shantha asked for this edit @shanthanordic @masz-nordic  Could you please clarify this?

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst (hunk 10 above)
Comment (FrancescoSer) [reply doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst:26]: reverted

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst (hunk 11 above)
Comment (FrancescoSer) [reply doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst:53]: Reverted

**Comment by FrancescoSer** [comment]
> This description looks like a copy of L15's but out of sync due to reviews in this PR. Any reason for not combining these two into common part and only listing differences between H20 and L15?

As I said on teams previously, FLPR on 54H is in scope of 2.8, while FLPR on 54L is not. When FLPR on 54L will be ready for public release, it will require docs and these could be published together and synced.

**Comment by FrancescoSer** [comment]
> This description looks like a copy of L15's but out of sync due to reviews in this PR. Any reason for not combining these two into common part and only listing differences between H20 and L15?

When SDP will go live for the next release, both this and the one for L15 should be moved in a more general section and synced.

**Comment by masz-nordic** [comment]
> When SDP will go live for the next release, both this and the one for L15 should be moved in a more general section and synced.

SDP is different from general FLPR usage, so those are two separate topics.
Pinging @wbober since he is aware of full vision for FLPR.

**Comment by FrancescoSer** [comment]
@gmarull @kylebonnici @masz-nordic Please rereview

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst (hunk 5 above)
Comment (kylebonnici) [reply doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_flpr.rst:127]: From the point of view of vscode The user can use create a separate build configuration and select flipr as the target.

In that case the command we run will be 

west build --build-dir <build> <app-dir> -pristine -target nrf54h20dk/nrf54h20/cpuflpr .....

One minor comment I why I sysbuild not being used? was sysbuil not created for such a case?

**Review by kylebonnici — APPROVED** [review]
VSCode part looks ok. This is not tested on our end and we have no requirement to support H20 for NCS 2.8

The test look good and on paper should work

**Review by annwoj — APPROVED** [review]

**Review by masz-nordic — APPROVED** [review]

**Review by b-gent — APPROVED** [review]
